src/physics/utilities/equation_solver.cpp

```diff
@@ -54,6 +64,24 @@ std::unique_ptr<mfem::IterativeSolver> EquationSolver::buildIterativeLinearSolve
   iter_lin_solver->SetMaxIter(lin_params.max_iter);
   iter_lin_solver->SetPrintLevel(lin_params.print_level);
 
+  // Handle the preconditioner - currently just BoomerAMG and HypreSmoother are supported
+  if (std::holds_alternative<HypreBoomerAMGPrec>(lin_params.prec)) {
+    auto par_fes = std::get<HypreBoomerAMGPrec>(lin_params.prec).pfes;
+    SLIC_ERROR_IF(par_fes == nullptr, "FESpace is required to use the HypreBoomerAMG preconditioner.");
```

Comment:
This won't exit until we change the logger to abortOnError and register the exit gracefully function with SLIC.

Reply:
I've gone ahead and changed the logger initialization to register exitGracefully so this will terminate as expected.